cmake/catkin_install_python.cmake

```diff
@@ -35,26 +35,47 @@ function(catkin_install_python signature)
         # write modified file with modified shebang line
         get_filename_component(python_name ${PYTHON_EXECUTABLE} NAME)
         string(REGEX REPLACE "${regex}" "#!/\\1/env ${python_name}\\2" data "${data}")
-        get_filename_component(filename ${file} NAME)
-        set(file "${CMAKE_CURRENT_BINARY_DIR}/catkin_generated/installspace")
-        file(MAKE_DIRECTORY ${file})
-        set(file "${file}/${filename}")
-        file(WRITE ${file} "${data}")
+        get_filename_component(filename ${source_file} NAME)
+        set(rewritten_file "${CMAKE_CURRENT_BINARY_DIR}/catkin_generated/installspace")
+        file(MAKE_DIRECTORY ${rewritten_file})
+        set(rewritten_file "${rewritten_file}/${filename}")
+        file(WRITE ${rewritten_file} "${data}")
       endif()
       # install (modified) file to destination
       set(optional_flag "")
       if(ARG_OPTIONAL)
         set(optional_flag "OPTIONAL")
       endif()
-      install(PROGRAMS "${file}" DESTINATION "${ARG_DESTINATION}" ${optional_flag})
+      # Install copy of file with re-written shebang to install space
+      install(PROGRAMS "${rewritten_file}" DESTINATION "${ARG_DESTINATION}" ${optional_flag})
 
-      get_filename_component(name "${file}" NAME)
+      # Hook for a platform specific wrapper around the modified python script
+      get_filename_component(name "${rewritten_file}" NAME)
       add_python_executable(SCRIPT_NAME ${name}
         # prefix with project name to avoid collisions across packages
         TARGET_NAME ${PROJECT_NAME}_${name}_exec_install_python
         DESTINATION "${ARG_DESTINATION}")
+
+      # Create devel-space wrapper if the destination is relative to the install prefix
+      if(NOT IS_ABSOLUTE ${ARG_DESTINATION})
+        message(STATUS "Installing devel-space wrapper ${source_file} to ${CATKIN_DEVEL_PREFIX}/${ARG_DESTINATION}")
+        # Create wrapper in devel space that uses source_file with but with correct shebang
+        set(PYTHON_SCRIPT ${source_file})
+        atomic_configure_file(${catkin_EXTRAS_DIR}/templates/script.py.in
+          ${CATKIN_DEVEL_PREFIX}/${ARG_DESTINATION}/${name}
+          @ONLY)
+
+        # Hook for a platform specific wrapper around the modified python script
+        get_filename_component(name "${source_file}" NAME)
```

Comment:
`name` should be the same as already determine above (currently line 53).

Reply:
Reuse `name` in 269a6aa